### PR TITLE
Update save type labels to use unambigous units

### PR DESCRIPTION
**Pull request by meeq**
## Description
Prefer `kbit` and `Mbit` instead of ambiguous `K` for save type units

## Motivation and Context

Reduces confusion, notably the inconsistency between EEPROM using kilobits and SRAM using kibibytes.

## How Has This Been Tested?
Tested on SummerCart64 with an NTSC console

## Types of changes

- [X] Improvement (non-breaking change that adds a new feature)
- [X] Bug fix (fixes an issue)
- [ ] Breaking change (breaking change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

Signed-off-by: meeq <me@christopherbonhage.com>

### CODE REVIEW

File: src/menu/views/load_rom.c [hunk 1]
```diff
@@ -165,12 +165,12 @@ static component_context_menu_t set_cic_type_context_menu = { .list = {
 static component_context_menu_t set_save_type_context_menu = { .list = {
     { .text = "Automatic", .action = set_save_type, .arg = (void *) (SAVE_TYPE_AUTOMATIC) },
     { .text = "None", .action = set_save_type, .arg = (void *) (SAVE_TYPE_NONE) },
-    { .text = "EEPROM 4kbit", .action = set_save_type, .arg = (void *) (SAVE_TYPE_EEPROM_4K) },
-    { .text = "EEPROM 16kbit", .action = set_save_type, .arg = (void *) (SAVE_TYPE_EEPROM_16K) },
-    { .text = "SRAM 256kbit", .action = set_save_type, .arg = (void *) (SAVE_TYPE_SRAM) },
+    { .text = "EEPROM 4kbit", .action = set_save_type, .arg = (void *) (SAVE_TYPE_EEPROM_4KBIT) },
+    { .text = "EEPROM 16kbit", .action = set_save_type, .arg = (void *) (SAVE_TYPE_EEPROM_16KBIT) },
+    { .text = "SRAM 256kbit", .action = set_save_type, .arg = (void *) (SAVE_TYPE_SRAM_256KBIT) },
     { .text = "SRAM 768kbit", .action = set_save_type, .arg = (void *) (SAVE_TYPE_SRAM_BANKED) },
```
Comment (Polprzewodnikowy) [inline src/menu/views/load_rom.c:171]: Shouldn't this be updated similarly to the label in the `format_rom_save_type`?

File: src/menu/rom_info.c [hunk 2]
```diff
@@ -147,234 +147,234 @@ typedef struct {
 static const match_t database[] = {
     MATCH_HOMEBREW_HEADER("ED"),                                                                            // Homebrew header (ED)
 
-    MATCH_CHECK_CODE(0x000000004CBC3B56, SAVE_TYPE_SRAM, FEAT_EXP_PAK_REQUIRED | FEAT_64DD_CONVERSION),     // DMTJ 64DD cartridge conversion
+    MATCH_CHECK_CODE(0x000000004CBC3B56, SAVE_TYPE_SRAM_256KBIT, FEAT_EXP_PAK_REQUIRED | FEAT_64DD_CONVERSION),     // DMTJ 64DD cartridge conversion
```
Comment (Polprzewodnikowy) [inline src/menu/rom_info.c:150]: Comments are no longer aligned

Screenshot 2024-06-11 114650

File: src/menu/views/load_rom.c (hunk 1 above)
Comment (meeq) [reply src/menu/views/load_rom.c]: https://github.com/Polprzewodnikowy/N64FlashcartMenu/pull/114/commits/bd6e4ceee5240ac5efb80b55a966f7b71579c9dd

File: src/menu/rom_info.c (hunk 2 above)
Comment (meeq) [reply src/menu/rom_info.c]: https://github.com/Polprzewodnikowy/N64FlashcartMenu/pull/114/commits/d47bcf964331ea386f65a3464d27224724cde2bd

**Pull request merged**